### ISSUE TITLE
Tightening role permissions?

### ISSUE THREAD

**Issue by sidshetye** [tartanguru/alexa-assistant#32]
Great work! You've described the role policy [here](https://github.com/tartanguru/alexa-assistant#aws-lambda-setup-part-1) in step "11" (the bullet numberings in markdown are messed up). There you can eliminate permissions for DynamoDB since it doesn't appear to be used anywhere ... i.e. remove this

```
        {
            "Effect": "Allow",
            "Action": "dynamodb:*",
            "Resource": "*"
        }
```
Also, the S3 bucket permissions can be tightened to something like
```
       {
            "Effect": "Allow",
            "Action": "s3:*",
            "Resource": "arn:aws:s3:::<bucket name>/*"
        }
```

Lastly - and this may need another ticket if it's longer - I didn't see a `package.json` or any build scripts, so are you handling node dependencies by hand and creating the `index.zip` file by hand too? Nothing wrong with it at this scale - just asking.

**Comment by tartanguru** [comment]
The intention was to use dynamoDB to store a users settings to allow
multi-tenancy on the skill.
Unfortunately there was a bug in the Alexa SDK at the time of publishing so
this feature was removed just be public release.
The permissions for DynamoDB have been left in to support this feature in a
future upgrade, as this will allow existing users to just upload a new zip.

With regards to the S3 bucket permission, you are indeed right that these could be much tighter, however this would have involved users being reasonably savvy about inserting the bucket name. Given the number of assistance requests I get for much simpler config items for the skill, I opted for the option that would give me the quietest life I'm afraid!

I haven't generated a package.json as conscious decision to make the package as easy to deploy as possible.

There are a lot of native binaries including gprc and LAME. These have to be built on an Amazon Linux machine in order for them to run in a Lambda function. Including them in the release ensures that the right binaries are present to deploy.
Additionally, the node_modules created by npm is pretty big which results in upload times to AWS of over 2 minutes which was slowing down development. A lot of this is from gprc and the Google API's so I stripped these down to the bare minimum for the Skill to run.

I will probably include a package.json when I update the skill following the release of the updated Assistant API
